src/plugins/auth/header.js

```diff
@@ -0,0 +1,22 @@
+"use strict";
+
+const Helper = require("../../helper");
+
+function headerAuth(manager, client, user, password, callback) {
+	// If no user is found, create it
+	if (!client) {
+		manager.addUser(user, null, true);
+	}
+
+	return callback(true);
+}
+
+function isHeaderAuthEnabled() {
```

Comment:
It seems like LDAP and Header Auth can be enabled at the same time. What happens in that case?

Reply:
In that case, the application should use the header-based authentication.

Reply:
ehm, why?
Why do you think header based auth trumps any other?

It should not. The only sane thing to do when you have both ldap and header auth enabled is to fail as that's clearly a nonsensically configuration

Reply:
unless you truly want to enable both, but is there a world where this really makes sesne?

Reply:
https://github.com/thelounge/thelounge/pull/4447#discussion_r795267583